src/sqlfluff/core/parser/grammar/greedy.py

```diff
@@ -161,6 +163,10 @@ def __init__(self, target, *args, **kwargs):
         self.target = self._resolve_ref(target)
         self.terminator = self._resolve_ref(kwargs.pop("terminator", None))
         self.include_terminator = kwargs.pop("include_terminator", False)
+
+        # StartsWith should only be used with a terminator
+        assert self.terminator
```

Comment:
Could we look at using this technique? "Required keyword argument". If it works, it seems perfect for this, because Python itself would enforce it, avoiding the need for an assertion or other runtime check.

Reply:
Had a look at this @barrywhart but not as easy if you want to also keep `*args` and `**kwargs`. Removing all of them on both `StartsWith` and the underlying `GreedyUntil` would be a bit of refactoring, that I think is beyond the scope of this PR.

You would also need to assign the `terminator` anyway so it would still be a line of code.

Finally I'm not convinced it's any clearer. In fact I think current code is clearer! And more in keeping with the rest of the style we use elsewhere.

So going to merge as is. Can revisit in a separate PR or in future if we want.